projects/plugins/jetpack/extensions/blocks/subscriptions/settings.js

```diff
@@ -88,13 +97,53 @@ export function NewsletterAccess( { accessLevel, setPostMeta, withModal = true }
 	}
 	const accessLabel = accessOptions[ accessLevel ]?.label;
 
+	const newsletterPlans = useSelect( select =>
+		select( 'jetpack/membership-products' )
+			?.getProducts()
+			?.filter( product => product.subscribe_as_site_subscriber )
+	);
+
 	// Can be “private”, “password”, or “public”.
 	const postVisibility = useSelect( select => select( 'core/editor' ).getEditedPostVisibility() );
 	const postVisibilityIsPublic = postVisibility === 'public';
 
 	const showVisibilityRestrictedMessage = ! postVisibilityIsPublic && accessLevel === 'everybody';
 	const showMisconfigurationMessage = ! postVisibilityIsPublic && accessLevel !== 'everybody';
 
+	if ( ! newsletterPlans || newsletterPlans.length === 0 ) {
+		return (
+			<>
+				<PanelRow>
+					{ __( 'Set up paid plan for readers to access your content.', 'jetpack' ) }
+				</PanelRow>
+				<PanelRow>
+					<ToolbarButton variant="primary" is href={ getPaidPlanLink( false ) } target="_blank">
```

Comment:
Maybe we could add an ExternalLink icon here as well, and remove the `is` attribute?

Reply:
Fixed in [869e9d5](https://github.com/Automattic/jetpack/pull/29989/commits/869e9d5d560e3f41a6eb726294ed8722a84ea537)